scripts/keys.coffee

```diff
@@ -22,81 +22,213 @@
+		keyname = msg.match[1]
 		name = msg.message.user.name 
 		user = robot.brain.userForName name
-		k = key()
-		if typeof user is 'object'
-			msg.send "Okay #{name} has keys"
-			k = user.name
-		robot.brain.set("key",k)	
-
+		if keyname is 'ravindra' or 'rv'
+			k = ravindra_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
+				k = user.name
+			robot.brain.set("ravindra_key",k)
+		else if keyname is 'kb' or 'kasturba'
+			k = kb_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
+				k = user.name
+			robot.brain.set("kb_key",k)
+		else if keyname is 'master'
+			k = root_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
+				k = user.name
+			robot.brain.set("master_key",k)	
 
-	robot.respond /i (don\'t|dont|do not) (has|have) (the key|key|keys|a key)/i , (msg)->
+#this section belongs to "i dont have key"
+	robot.respond /i (don\'t|dont|do not) (has|have) (.+) (the key|key|keys|a key)/i , (msg)->
+		keyname = msg.match[3]
 		name = msg.message.user.name
 		user = robot.brain.userForName name
-		k = key()
-		if k is name
-			k = ""
-		if typeof user is 'object'
-			if k is ""
-				msg.send "Okay #{name} doesn't have keys. Who got the keys then?"
-			else
-				msg.send "Yes , i know buddy, its because #{k} has got the keys"	
-		robot.brain.set("key",k)	
-
+		
+		if keyname is 'ravindra' || keyname is 'rv'
+			k = ravindra_key()
+			if k is name
+				k = ""
+			if typeof user is 'object'
+				if k is ""
+					msg.send "Okay #{name} doesn't have ravindra keys. Who got the ravindra keys then?"
+				else
+					msg.send "Yes , i know buddy, its because #{k} has got the ravindra keys"	
+			robot.brain.set("ravindra_key",k)	
+		else if keyname is 'kb' || keyname is 'kasturba'
+			k = kb_key()
+			if k is name
+				k = ""
+			if typeof user is 'object'
+				if k is ""
+					msg.send "Okay #{name} doesn't have kb keys. Who got the kb keys then?"
+				else
+					msg.send "Yes , i know buddy, its because #{k} has got the kb keys"	
+			robot.brain.set("kb_key",k)	
+		else if keyname is 'master'
+			k = master_key()
+			if k is name
+				k = ""
+			if typeof user is 'object'
+				if k is ""
+					msg.send "Okay #{name} doesn't have master keys. Who got the master keys then?"
+				else
+					msg.send "Yes , i know buddy, its because #{k} has got the master keys"	
+			robot.brain.set("master_key",k)	
 
-	robot.respond /(.+) (has|have) (the key|key|keys|a key)/i , (msg)->
+#this section belongs to "$name has keys"
+	robot.respond /(.+) (has|have) (.+) (the key|key|keys|a key)/i , (msg)->
 		othername = msg.match[1]
+		keyname = msg.match[3]
 		name = msg.message.user.name
-		k = key()
-		unless othername in ["who", "who all","Who", "Who all" , "i" , "I" , "i don't" , "i dont" , "i do not" , "I don't" , "I dont" , "I do not"]
+		
+		if keyname is 'ravindra' || keyname is 'rv'
+			k = ravindra_key()
```

Comment:
This also looks like the code is being repeated. If you find a solution to remove repetition in the above lines then a similar strategy will work here too.